ui/src/conceptList.tsx

```diff
@@ -0,0 +1,103 @@
+import Typography from "@mui/material/Typography";
+import { DataGrid, GridColDef } from "@mui/x-data-grid";
+import React, { useContext, useEffect, useMemo, useState } from "react";
+import { useParams } from "react-router-dom";
+import { EntityInstancesApiContext } from "./apiContext";
+import { State } from "./state";
+
+// Row acts as a bridge between the data format returned from the API and the
+// format expected by DataGrid.
+class Row {
+  id = 0;
+  [x: string]: string | number | boolean | undefined;
+}
+
+type ConceptListProps = {
+  underlay: string;
+  state: State;
+  filter: string;
+};
+
+export default function ConceptList(props: ConceptListProps) {
+  const [error, setError] = useState<Error | null>(null);
+  const [rows, setRows] = useState<Array<Row> | null>(null);
+
+  const params = useParams<{ group: string; criteria: string }>();
+  const api = useContext(EntityInstancesApiContext);
+
+  const columns: GridColDef[] = useMemo(() => {
+    return [
+      { field: "concept_name", headerName: "Concept Name", width: 400 },
+      { field: "concept_id", headerName: "Concept ID", width: 100 },
+      { field: "domain_id", headerName: "Domain", width: 100 },
+    ];
+  }, []);
+
+  useEffect(() => {
+    api
+      .searchEntityInstances({
+        entityName: "concept",
+        underlayName: props.underlay,
+        searchEntityInstancesRequest: {
+          entityDataset: {
+            entityVariable: "c",
+            selectedAttributes: columns.map((col) => {
+              return col.field;
+            }),
+            filter: {
+              relationshipFilter: {
+                outerVariable: "c",
+                newVariable: "cc",
+                newEntity: props.filter,
+              },
+            },
+          },
+        },
+      })
+      .then(
+        (res) => {
+          const rows: Row[] = [];
+          if (!!res.instances) {
```

Comment:
nit: technically !! is unnecessary here, as this expression would already be coerced to a boolean given the context